app/src/main/java/de/westnordost/streetcomplete/quests/construction/MarkCompletedConstruction.java

```diff
@@ -0,0 +1,76 @@
+package de.westnordost.streetcomplete.quests.construction;
+
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+
+import de.westnordost.osmapi.map.data.Element;
+import de.westnordost.streetcomplete.R;
+import de.westnordost.streetcomplete.data.meta.OsmTaggings;
+import de.westnordost.streetcomplete.data.osm.Countries;
+import de.westnordost.streetcomplete.data.osm.OsmElementQuestType;
+import de.westnordost.streetcomplete.data.osm.changes.StringMapChangesBuilder;
+import de.westnordost.streetcomplete.data.osm.download.OverpassMapDataDao;
+import de.westnordost.streetcomplete.quests.AbstractQuestAnswerFragment;
+import de.westnordost.streetcomplete.quests.DateHandler;
+import de.westnordost.streetcomplete.quests.YesNoQuestAnswerFragment;
+
+public abstract class MarkCompletedConstruction implements OsmElementQuestType {
+	protected final OverpassMapDataDao overpassServer;
+
+	public MarkCompletedConstruction(OverpassMapDataDao overpassServer)
+	{
+		this.overpassServer = overpassServer;
+	}
+
+	@Nullable
+	@Override public Boolean isApplicableTo(Element element)
+	{
+		/* Whether this element applies to this quest cannot be determined by looking at that
+		   element alone (see download()), an Overpass query would need to be made to find this out.
+		   This is too heavy-weight for this method so it always returns false. */
+
+		/* The implications of this are that this quest will never be created directly
+		   as consequence of solving another quest and also after reverting an input,
+		   the quest will not immediately pop up again. Instead, they are downloaded well after an
+		   element became fit for this quest. */
+		return null;
+	}
+
+	public AbstractQuestAnswerFragment createForm()
+	{
+		return new YesNoQuestAnswerFragment();
+	}
+
+	public static String getQueryPart(String key, String nameOfGeneratedGroup, int rewievIntervalInDays){
+		String currentDate = DateHandler.getCurrentDateString() + "T00:00:00Z";
+		String repeatedReviewCutoffDate = DateHandler.getOffsetDateString(-rewievIntervalInDays) + "T00:00:00Z";
+
+		return
+			"way[" + key + "=construction] -> .under_construction;" +
+				"(" +
+				"way[" + key + "=construction][opening_date](" +
+				"  if:is_date(t['opening_date']) && date(t['opening_date'])>date('" + currentDate + "'));" +
+				") -> .known_opening_date_in_future;" +
+				"(" +
+				"  way[" + key + "=construction](newer:'" + repeatedReviewCutoffDate + "');" +
+				") -> .recently_edited;" +
+				"((" +
+				"	.under_construction" +
+				"	- .known_opening_date_in_future)" +
+				"	- .recently_edited" +
+				") -> " + nameOfGeneratedGroup + ";";
+	}
```

Comment:
This query could probably be simplified? Perhaps i.e. like this(?):
`way[highway=construction](!newer: my-cutoff-date)(if:!is_date(t['opening_date']) || date(t['opening_date'])<date('" + currentDate + "'))`

Reply:
"Did you test that this works?", obviously yes, before testing it was full of bugs of various kinds (from invalid queries to not handling unexpected tagging - it even ended with a [new tiny feature for overpass turbo](https://wiki.openstreetmap.org/w/index.php?title=Overpass_turbo/Extended_Overpass_Turbo_Queries&diff=1570006&oldid=1564679) :)  ).

I was not even thinking about setting up unit test, but it may be a good idea - with query complexity it will be useful as documentation. Main problem is that unit tests rely on OSM data and this kind of data will be especially fragile. But maybe for unit tests we should query historical data that will not change? I just though about it, but it seems that it should solve problem of fragile tests, at cost of even greater complexity.

It is likely that it can be simplified, but I will try to set up unit test before that.

Reply:
Cool! Well, I was just wondering about the (unit) tests. If to enable testing at all results in (much) more complex code, one should think twice if it is worth it, in my opinion.
You are right of course that this construction data is especially fragile.... perhaps we could use the Airport Berlin as a relatively stable construction site, haha (construction began in 2006, not ready until at least 2020)

Reply:
Yes, much more complex code is not worth it - but hopefully it can be done without it.

Reply:
`!newer` - unfortunately neither `older` or `!newer` is an allowed Overpass Turbo syntax. I used other ideas to simplify the query, thanks! 

See https://github.com/tyrasd/overpass-turbo/issues/247 for a related feature request